shared/common-adapters/tab-bar.native.js

```diff
@@ -29,6 +29,18 @@ class SimpleTabBarButton extends Component<void, ItemProps, void> {
   }
 }
 
+const BadgeComponent = ({badgeNumber, badgePosition}: {badgeNumber: number, badgePosition: TabBadgePosition}) => {
```

Comment:
I don't get what you mean by passing badge style, the top-right position needs absolute positioning (within another Box component) and the default doesn't. And I need an if, else somewhere.